design/specs/column-extraction.md

```diff
@@ -8,87 +8,62 @@ editor: markdown
+In data modeling, it's usual practice to split data up into related tables. In the context of Mathesar, one way of doing this is by extracting columns from a table into a new table. This process is known as column extraction.
 
-## Selecting Columns
-Columns must be given a new interaction in order to be selected. A column header can be selected by clicking on it, as in spreadsheet applications. Multi-selection is also possible by holding down the shift key and clicking on multiple column headers. Clearing the selection is as simple as clicking anywhere outside of the selection target or on a selected column once more.
+This spec describes the steps a user would take to create a new table from a subset of columns from an existing table.
 
-## Selection Actions
-Depending on a particular selection, the inspector panel will display actions that can be performed on the selection, as described above with columns. These selection actions will be determined by the type and number of items chosen.
+## Design Goals
 
-# User Flow
+- Allow users to split a table into multiple new tables by selecting columns
+- Help users understand the impact of splitting a table
+- Help users understand how the tables are linked after the split operation
 
-1. **The user imports a table from a file**
+## Splitting a Table by Extracting Columns
 
-    This user flow assumes that the starting point is an imported table containing columns that can be extracted and turned into entities. A column that may be transformed into a table or entity will usually include other columns that are attributes for that entity. If the database is not normalized, the entity to be extracted might include duplicated values. For example, consider a table of publications with multiple entries for the same publisher in the publisher column.
+### Selecting columns as a starting point
 
-2. **The user selects the columns (attributes) for the new entity**
+The user will select one or multiple columns to start with the column move process. All columns can be moved, except for primary key columns, unorderable columns like JSON-type columns and referent columns (Columns referenced by other columns).
 
-    The user chooses all the columns that will be attributes for the new object. For example, in the instance of the publisher, there could be columns for 'Publisher Name' and 'Publisher Id.' Including all the attributes is very important since columns cannot be moved after the fact.
+Once selected, the inspector panel will list 'New linked table from columns' as an action. Selecting this option will open the 'New linked table from columns' dialog.
 
-     <img width="1512" alt="image" src="https://user-images.githubusercontent.com/845767/177544077-9f074830-ce64-4ed9-80bc-53de53977f9c.png">
+When a user selects a column that cannot be moved, the move action will not be available in the inspector panel.
```

Comment:
Can we instead disable the action and have a help icon that explains why it's disabled?

Reply:
Yes, I think that will work. I have updated